src/backend/services/request_validators.py

```diff
@@ -36,47 +33,43 @@ def validate_deployment_model(deployment: str, model: str, session: DBSessionDep
         HTTPException: If the deployment and model are not compatible
 
     """
-    deployment_db = deployment_crud.get_deployment_by_name(session, deployment)
-    if not deployment_db:
-        deployment_db = deployment_crud.get_deployment(session, deployment)
-
-    # Check deployment config settings availability
-    deployment_config = find_config_by_deployment_id(deployment)
-    if not deployment_config:
-        deployment_config = find_config_by_deployment_name(deployment)
-    if not deployment_config:
+    found = deployment_service.get_deployment_definition_by_name(session, deployment)
+    if not found:
+        found = deployment_service.get_deployment_definition(session, deployment)
+    if not found:
         raise HTTPException(
             status_code=400,
             detail=f"Deployment {deployment} not found or is not available in the Database.",
         )
 
-    if not deployment_db:
-        deployment_db = deployment_crud.create_deployment_by_config(session, deployment_config)
-    if not deployment_db:
-        raise HTTPException(
-            status_code=400,
-            detail=f"Deployment {deployment} not found or is not available in the Database.",
-        )
     # Validate model
+    # deployment_model = next(
+    #     (
+    #         model_db
+    #         for model_db in found.models
+    #         if model_db.name == model or model_db.id == model
+    #     ),
+    #     None,
+    # )
     deployment_model = next(
         (
             model_db
-            for model_db in deployment_db.models
-            if model_db.name == model or model_db.id == model
+            for model_db in found.models
+            if model_db == model
         ),
         None,
     )
     if not deployment_model:
         deployment_model = model_crud.create_model_by_config(
-            session, deployment_db, deployment_config, model
+            session, found, deployment_config, model
         )
```

Comment:
As I see, deployment_config has been removed, and I don’t completely understand why.The idea is that if, for some reason, there is no deployment or model in the database, we attempt to recreate it using a predefined configuration. It was also used during the agent creation flow